docker/Dockerfile

```diff
@@ -0,0 +1,73 @@
+# This Dockerfile creates a production release image for the project. This
+# downloads the release from releases.hashicorp.com and therefore requires that
+# the release is published before building the Docker image.
+#
+# We don't rebuild the software because we want the exact checksums and
+# binary signatures to match the software and our builds aren't fully
+# reproducible currently.
+ARG BASE_ARCH
```

Comment:
We don't need these two lines here anymore since we are using `docker buildx` in the release process rather than building 4 architecture images and then using `docker manifest`.

Reply:
:+1: